presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -857,10 +857,19 @@ protected Object visitNotExpression(NotExpression node, Object context)
         protected Object visitLogicalBinaryExpression(LogicalBinaryExpression node, Object context)
         {
             Object left = process(node.getLeft(), context);
-            Object right = process(node.getRight(), context);
+            Object right;
 
             switch (node.getType()) {
                 case AND: {
+                    if (Boolean.FALSE.equals(left)) {
+                        return false;
+                    }
+
+                    right = process(node.getRight(), context);
+
+                    if (node.getType() == LogicalBinaryExpression.Type.AND && Boolean.FALSE.equals(left)) {
```

Comment:
`node.getType() == LogicalBinaryExpression.Type.AND` you already know that :)
`Boolean.FALSE.equals(left)` you already know the negation of this :)